pycalphad/core/workspace.py

```diff
@@ -254,25 +270,40 @@ class Workspace:
     database: Database = TypedField(lambda _: None)
     components: SpeciesList = ComponentsField(depends_on=['database'])
     phases: PhaseList = PhasesField(depends_on=['database', 'components'])
-    conditions: Conditions = ConditionsField(lambda wks: Conditions(wks))
+    conditions: Conditions = ConditionsField(lambda wks: Conditions(wks), depends_on=['components', 'models'])
     verbose: bool = TypedField(lambda _: False)
     models: Mapping[PhaseName, ModelType] = ModelsField(depends_on=['phases', 'parameters'])
     parameters: SumType([NoneType, Dict]) = DictField(lambda _: OrderedDict())
     phase_record_factory: Optional[PhaseRecordFactory] = PRFField(depends_on=['phases', 'conditions', 'models', 'parameters'])
     calc_opts: SumType([NoneType, Dict]) = DictField(lambda _: OrderedDict())
     solver: SolverBase = SolverField(lambda obj: Solver(verbose=obj.verbose), depends_on=['verbose'])
     # eq is set by a callback in the EquilibriumCalculationField (TypedField)
-    eq: Optional[LightDataset] = EquilibriumCalculationField(depends_on=['phase_record_factory', 'calc_opts', 'solver'])
+    eq: Optional[LightDataset] = EquilibriumCalculationField(depends_on=['phase_record_factory', 'conditions', 'calc_opts', 'solver'])
 
     def __init__(self, *args, **kwargs):
```

Comment:
Maybe a little unrelated, but should we change the signature to 

```suggestion
    def __init__(self, database=None, components=None, phases=None, conditions=None, *, **kwargs):
```

right now we're throwing away any positional arguments past the first four. Per [PEP-3102](https://peps.python.org/pep-3102/), we can syntactically disallow any positional arguments beyond the first four

Reply:
This turns out to be a SyntaxError as `**kwargs` is not allowed after a bare `*`